src/components/layout/ToastNotification.tsx

```diff
@@ -1,32 +1,75 @@
 import React from 'react'
-import { Alert, Snackbar, Typography } from '@mui/material'
+import {
+  Alert,
+  IconButton,
+  InputAdornment,
+  Snackbar,
+  Stack,
+  TextField,
+  Typography,
+} from '@mui/material'
 import { useToastNotification, useToastNotificationStore } from '@/stores'
-import { Trans } from 'react-i18next'
+import { useTranslation } from 'react-i18next'
+import ContentCopyIcon from '@mui/icons-material/ContentCopy'
 
 const _ToastNotification: React.FC = () => {
   const { hideToast } = useToastNotification()
   const isShown = useToastNotificationStore((state) => state.isShown)
   const message = useToastNotificationStore((state) => state.message)
   const severity = useToastNotificationStore((state) => state.severity)
+  const correlationId = useToastNotificationStore((state) => state.correlationId)
+  const { t } = useTranslation('error')
 
   const id = React.useId()
 
+  const correlationIdSection = (
+    <>
+      <p>{t('axiosError.correlationIdText')}</p>
+      <TextField
+        multiline
+        id="outlined-read-only-input"
+        label="Correlation ID"
+        defaultValue={correlationId}
+        InputProps={{
+          readOnly: true,
+          endAdornment: (
+            <InputAdornment position="end">
+              <IconButton onClick={() => handleCopyCorrelationId(correlationId)}>
+                <ContentCopyIcon />
+              </IconButton>
+            </InputAdornment>
+          ),
+        }}
+      />
+    </>
+  )
+
+  async function handleCopyCorrelationId(correlationId: string | undefined) {
+    if (!correlationId) return
+    try {
+      await navigator.clipboard.writeText(correlationId)
+    } catch (error) {
+      console.error('Unable to copy the correlationId:', error)
+    }
+  }
+
   return (
     <Snackbar
       open={isShown}
-      onClose={hideToast}
       autoHideDuration={5000}
       anchorOrigin={{ vertical: 'bottom', horizontal: 'right' }}
-      sx={{ maxWidth: 480 }}
+      sx={{ maxWidth: 350 }}
```

Comment:
Any reason why the `maxWidth` is reduced?

Reply:
> Any reason why the `maxWidth` is reduced?

It's reduced to allow the snackbar to remain inside the drawer as in the design